File: java/2015/day02a/Program.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.Vector;

public class Program {
	public static void usage() {
		System.err.println("usage: java -jar Program.jar <input file>");
		System.exit(1);
	}

	public static int process(String filename) {
		int totalArea = 0;
		try ( BufferedReader reader = new BufferedReader(new FileReader(filename))) {
			String line;
			while ( (line = reader.readLine()) != null ) {
				String[] numStrs = line.split("x");
				Vector<Integer> dimensions = new Vector<Integer>();
				for ( String numStr : numStrs ) {
					dimensions.addElement(Integer.parseInt(numStr));
				}
				int area1 = dimensions.elementAt(0) * dimensions.elementAt(1);
				int area2 = dimensions.elementAt(0) * dimensions.elementAt(2);
				int area3 = dimensions.elementAt(1) * dimensions.elementAt(2);
				int surfaceArea = area1 * 2 + area2 * 2 + area3 * 2;
				int minArea = Integer.min(Integer.min(area1, area2), area3);
				totalArea += surfaceArea + minArea;
			}
		} catch ( IOException e ) {
			System.err.println("Error reading file: " + e.getMessage());
			System.exit(1);
		}
		return totalArea;
	}

	public static final void main(String[] args) {
		if ( args.length < 1 ) {
			usage();
		}
		String filename = args[0];
		int result = process(filename);
		System.out.println("result = " + result);
	}
}

File: kotlin/2015/day02a/Program.kt
import java.io.File

fun usage() {
    System.err.println("usage: java -jar Program.jar <input file>")
    kotlin.system.exitProcess(1)
}

fun process(content: String) : Int {
    var totalArea = 0

    for ( line in content.split("\n") ) {
        if ( !line.isEmpty() ) {
            val dims = line.split("x").map { it.toInt() }
            val l = dims[0]
            val w = dims[1]
            val h = dims[2]
            val area1 = l * w
            val area2 = l * h
            val area3 = w * h
            val surfaceArea = 2 * area1 + 2 * area2 + 2 * area3
            val minArea = listOf(area1, area2, area3).minOrNull() ?: 0
            totalArea += surfaceArea + minArea
        }
    }

    return totalArea
}

fun main(args: Array<String>) {
    if ( args.isEmpty()) {
        usage()
    }
    val filename = args[0]
    try {
        val content = File(filename).readText()
        val result = process(content)
        println("result = $result")
    } catch ( e: Exception ) {
        System.err.println("error processing file: ${e.message}")
    }
}
